core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java

```diff
@@ -56,14 +56,14 @@
 public class AddColumnTask
         implements DataDefinitionTask<AddColumn>
 {
-    private final Metadata metadata;
+    private final PlannerContext plannerContext;
     private final AccessControl accessControl;
     private final ColumnPropertyManager columnPropertyManager;
 
     @Inject
-    public AddColumnTask(Metadata metadata, AccessControl accessControl, ColumnPropertyManager columnPropertyManager)
+    public AddColumnTask(PlannerContext plannerContext, AccessControl accessControl, ColumnPropertyManager columnPropertyManager)
     {
-        this.metadata = requireNonNull(metadata, "metadata is null");
```

Comment:
Does it make sense to extract `Metadata` in the constructor for all of these, since that's the only field they use?

Reply:
All of these tasks actually use the planner context.  The downstream services need it.